lib/Optimizer/Dialect/CC/CCOps.cpp

```diff
@@ -966,6 +966,16 @@ struct FuseWithConstantArray
 
           return success();
         }
+        if (auto complexTy = dyn_cast<ComplexType>(extval.getType())) {
+          std::int32_t i = extval.getRawConstantIndices()[0];
+          auto rePart = cast<FloatAttr>(conarr.getConstantValues()[2 * i]);
+          auto imPart = cast<FloatAttr>(conarr.getConstantValues()[2 * i + 1]);
```

Comment:
I think the default checks make sure the index is within the range of the size of the array. So that check may not be good enough when accessing `2 * i`. An additional check here would be needed in that case. We shouldn't make any assumptions here and risk crashing with a SIGSEGV.